### PR TITLE
SONAR-8664 add organization to response of api/notifications/list

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/notification/ws/ListAction.java [hunk 1]
```diff
@@ -146,35 +150,53 @@ private boolean isDispatcherAuthorized(PropertyDto prop, String dispatcher) {
...
-    return dbClient.componentDao().selectByIds(dbSession,
-      properties.stream()
-        .filter(prop -> prop.getResourceId() != null)
-        .map(PropertyDto::getResourceId)
-        .distinct()
-        .collect(toList()))
+    Set<String> authorizedComponentUuids = new HashSet<>(dbClient.authorizationDao().selectAuthorizedRootProjectsUuids(dbSession, userSession.getUserId(), UserRole.USER));
+    Set<Long> componentIds = properties.stream()
+      .filter(prop -> prop.getResourceId() != null)
+      .map(PropertyDto::getResourceId)
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/notification/ws/ListAction.java]: could also be written `.map(PropertyDto::getResourceId).filter(Objects::nonNull)` to remove the condition `!=null` :-)

File: sonar-ws/src/main/protobuf/ws-notifications.proto [hunk 2]
```diff
@@ -35,6 +35,7 @@ message ListResponse {
 message Notification {
   optional string channel = 1;
   optional string type = 2;
+  optional string organization = 5;
```
Comment (simonbrandhof) [inline sonar-ws/src/main/protobuf/ws-notifications.proto]: having 5 before 3 and 4 is error-prone for future additions.

Comment (sns-seb) [inline sonar-ws/src/main/protobuf/ws-notifications.proto]: but the error doesn't last past the compilation, so...

I think it's a tradeoff between backward compatibility and field sorting that functionally makes sense

File: server/sonar-server/src/main/java/org/sonar/server/notification/ws/ListAction.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/notification/ws/ListAction.java]: legacy code, but yeah, I can improve it

File: sonar-ws/src/main/protobuf/ws-notifications.proto (hunk 2 above)
Comment (simonbrandhof) [inline sonar-ws/src/main/protobuf/ws-notifications.proto]: I was just talking about moving the line, not changing the field number !

Comment (sns-seb) [inline sonar-ws/src/main/protobuf/ws-notifications.proto]: that's my point, functionally it makes more sense to have the organization before the component (key), ie. sort from the most generic to the most specific. Yet, organization can be number '3' for backward compatibility reasons.

**Pull request merged**